chia/rpc/full_node_rpc_api.py

```diff
@@ -397,6 +405,48 @@ async def get_block_records(self, request: Dict) -> Optional[Dict]:
             records.append(record)
         return {"block_records": records}
 
+    async def get_block_spends(self, request: Dict) -> Optional[Dict]:
+        if "header_hash" not in request:
+            raise ValueError("No header_hash in request")
+        header_hash = bytes32.from_hexstr(request["header_hash"])
+        full_block: Optional[FullBlock] = await self.service.block_store.get_full_block(header_hash)
+        if full_block is None:
+            raise ValueError(f"Block {header_hash.hex()} not found")
+
+        tx_info: Optional[TransactionsInfo] = full_block.transactions_info
```

Comment:
as far as I can tell, the only reason you pull out `tx_info` is to get to the `cost` field. Since this block can be assumed to have been validated already, I don't think we need to do this. We could just use the `MAX_BLOCK_COST_CLVM` as the upper bound, but we already know the program won't exceed that.